ColorPatterns/Helpers/Starling.swift

```diff
@@ -0,0 +1,266 @@
+                let newPlayer = createNewPlayerAttachedToEngine()
+                players.append(newPlayer)
+                return newPlayer
+            }
+            return player
+        }()
+
+        return player
+    }
+
+    private func createNewPlayerAttachedToEngine() -> StarlingAudioPlayer {
+        let player = StarlingAudioPlayer()
+        engine.attach(player.node)
+        engine.connect(player.node, to: engine.mainMixerNode, format: nil)
+        return player
+    }
+
+    private func didFinishLoadingAudioFile(_ file: AVAudioFile, identifier: SoundIdentifier) {
+        // Note: self is used as the lock pointer here to avoid
+        // the possibility of locking on _swiftEmptyDictionaryStorage
+        objc_sync_enter(self)
+        files[identifier] = file
+        objc_sync_exit(self)
+    }
+
+    private func handleNonFatalError(_ error: Error) {
+        print("*** Starling error: \(error)")
+    }
+
+    // MARK: - Debugging / Diagnostics
+
+    private var diagnosticTimer: Timer?
+    public func beginPeriodicDiagnostics() {
+        guard diagnosticTimer == nil else { return }
+        diagnosticTimer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(diagnosticTimerFire(_:)), userInfo: nil, repeats: true)
+    }
+
+    public func stopDiagnostics() {
+        diagnosticTimer?.invalidate()
+        diagnosticTimer = nil
+    }
+
+    @objc private func diagnosticTimerFire(_ timer: Timer) {
+        print("****** Starling Debug Info ******")
+        print("Audio files loaded: \(files.count)")
+        objc_sync_enter(players)
+        print("Total living players: \(players.count)")
+        print("Currently playing: \(players.filter({ $0.state.status != .idle }).count)")
+        for (index, player) in players.enumerated() {
+            print("Player \(index): \(player.state.status == .idle ? "Idle" : "Playing \(player.state.sound ?? "(Null)")")")
+        }
+        print("\n")
+        objc_sync_exit(players)
+    }
+}
+
+/// The internal playback status. This is somewhat redundant at the moment given that
+/// it should effectively mimic -isPlaying on the node, however it may be extended in
+/// the future to represent other non-binary playback modes.
+///
+/// - idle: No sound is currently playing.
+/// - playing: A sound is playing.
+/// - looping: (Not currently implemented)
+private enum PlayerStatus {
+    case idle
+    case playing
+    case looping
+}
+
+private struct PlayerState {
+    let sound: SoundIdentifier?
+    let status: PlayerStatus
+
+    static func idle() -> PlayerState {
+        return PlayerState(sound: nil, status: .idle)
+    }
+}
+
+private class StarlingAudioPlayer {
+    let node = AVAudioPlayerNode()
+    var state: PlayerState = PlayerState.idle()
+
+    func play(_ file: AVAudioFile, identifier: SoundIdentifier) {
+        node.scheduleFile(file, at: nil, completionCallbackType: .dataPlayedBack) { [weak self] callbackType in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/vandermesis/ColorPatterns/blob/8c9d8b0cbd258e51fec07cc9cd8e5c65cf9e26ea/ColorPatterns/Helpers/Starling.swift#L243"></span>Unused parameter “callbackType” in a closure should be replaced with _.<br />
<code>unused_closure_parameter</code> <code>Starling.swift:243</code></td>
    </tr>
  </tbody>
</table>